sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,348 @@
+            }
+        }
+    }
+
+    /// Sends a read receipt.
+    /// - Parameters:
+    ///   - request: Request for sending a read receipt.
+    ///   - options: Send read receipt options.
+    ///   - completionHandler: A completion handler that receives a status code on success.
+    public func send(
+        readReceipt: SendReadReceiptRequest,
+        withOptions options: AzureCommunicationChatService.SendChatReadReceiptOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        service
+            .send(chatReadReceipt: readReceipt, chatThreadId: threadId, withOptions: options) { result, httpResponse in
+                switch result {
+                case .success:
+                    completionHandler(.success(()), httpResponse)
+
+                case let .failure(error):
+                    completionHandler(.failure(error), httpResponse)
+                }
+            }
+    }
+
+    /// Lists read receipts for the ChatThread.
+    /// - Parameters:
+    ///   - options: List chat read receipts options.
+    ///   - completionHandler: A completion handler that receives the list of read receipts on success.
+    public func listReadReceipts(
+        withOptions options: AzureCommunicationChatService.ListChatReadReceiptsOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<PagedCollection<ChatMessageReadReceipt>>
+    ) {
+        service.listChatReadReceipts(chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case let .success(readReceipts):
+                completionHandler(.success(readReceipts), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a typing notification.
+    /// - Parameters:
+    ///    - options: Send typing notification options
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func sendTypingNotification(
+        withOptions options: AzureCommunicationChatService.SendTypingNotificationOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        service.sendTypingNotification(chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case .success:
+                completionHandler(.success(()), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a message to a ChatThread.
+    /// - Parameters:
+    ///    - chatMessage : Request for sending a message.
+    ///    - options: A list of options for the operation.
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func send(
+        message: SendChatMessageRequest,
```

Comment:
Consider accepting just `ChatMessage` here.

Reply:
For this, SendReadReceiptRequest, and CreateChatThreadRequest will leave as is for similar reasons - ChatThread, ReadReceipt and ChatMessage contains extra properties